sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala

```diff
@@ -83,8 +80,10 @@ private[sql] class JSONRelation(
     val job = new Job(sqlContext.sparkContext.hadoopConfiguration)
     val conf = SparkHadoopUtil.get.getConfigurationFromJobContext(job)
 
-    val paths = inputPaths.map(_.getPath)
-
+    var paths = inputPaths.map(_.getPath)
+    if (paths.isEmpty) {
+      paths = cachedSourcePaths()   // try with original sources
```

Comment:
Just wanted to delegate making exception to underlying input formatter. Anyway, it's resolved and I'll closed this PR. Thanks.